Assets/Scripts/Game/Automap.cs

```diff
@@ -2481,11 +2481,15 @@ private void RestoreStateAutomapDungeon(bool forceNotVisitedInThisRun = false)
                     {
                         Transform currentTransformModel = currentTransformElement.GetChild(indexModel);
 
-                        MeshRenderer meshRenderer = currentTransformModel.GetComponent<MeshRenderer>();
-                        if (meshRenderer)
-                        {
-                            UpdateMeshRendererDungeonState(ref meshRenderer, automapDungeonState, indexBlock, indexElement, indexModel, forceNotVisitedInThisRun);
-                        }
+						MeshRenderer[] meshRenderers = currentTransformModel.GetComponentsInChildren<MeshRenderer>();
+						foreach (MeshRenderer meshRendererIteration in meshRenderers)
+						{
+							if (meshRendererIteration)
+							{
+								MeshRenderer meshRenderer = meshRendererIteration;
+								UpdateMeshRendererDungeonState(ref meshRenderer, automapDungeonState, indexBlock, indexElement, indexModel, forceNotVisitedInThisRun);
```

Comment:
Passing by reference here is unneeded, you can change method signature if @Nystul-the-Magician agrees.

Reply:
no objections from my side ;)

Reply:
Done.